test/unittest/tests/block/test_block_session.cpp

```diff
@@ -0,0 +1,281 @@
+        ext->next[0] = ext2;
+        bms->ext_cache = ext;
+        bms->ext_cache_cnt = 2;
+
+        WT_EXT *cached_ext;
+        REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &cached_ext) == 0);
+        REQUIRE(ext == cached_ext);
+        REQUIRE(ext2 != cached_ext);
+        validate_and_free_ext_list(bms, 1);
+        validate_and_free_ext_block(cached_ext);
+    }
+}
+
+TEST_CASE("Block session: __wti_block_ext_free", "[block_session]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    WT_BLOCK_MGR_SESSION *bms = session->setupBlockManagerSession();
+
+    /*
+     * FIXME-WT-13451: Update __wti_block_ext_free function to test that block is set to null
+     * SECTION("Free with null block manager session")
+     * {
+     *    std::shared_ptr<MockSession> session_no_bm = MockSession::buildTestMockSession();
+     *    WT_EXT *ext;
+     *
+     *    REQUIRE(__ut_block_ext_alloc(session_no_bm->getWtSessionImpl(), &ext) == 0);
+     *    REQUIRE(ext != nullptr);
+     *
+     *    __wti_block_ext_free(session_no_bm->getWtSessionImpl(), ext);
+     *    REQUIRE(ext == nullptr);
+     * }
+     */
+
+    SECTION("Calling free with cache")
+    {
+        WT_EXT *ext;
+        REQUIRE(__ut_block_ext_alloc(session->getWtSessionImpl(), &ext) == 0);
+
+        __wti_block_ext_free(session->getWtSessionImpl(), ext);
+
+        REQUIRE(ext != nullptr);
+        REQUIRE(bms->ext_cache == ext);
+        validate_ext_list(bms, 1);
+
+        WT_EXT *ext2;
+        REQUIRE(__ut_block_ext_alloc(session->getWtSessionImpl(), &ext2) == 0);
+        __wti_block_ext_free(session->getWtSessionImpl(), ext2);
+
+        REQUIRE(ext != nullptr);
+        REQUIRE(bms->ext_cache == ext2);
+        REQUIRE(bms->ext_cache->next[0] == ext);
+        validate_and_free_ext_list(bms, 2);
+    }
+}
+
+TEST_CASE("Block session: __block_ext_discard", "[block_session]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    WT_BLOCK_MGR_SESSION *bms = session->setupBlockManagerSession();
+
+    WT_EXT *ext, *ext2, *ext3;
+    REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &ext) == 0);
+    REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &ext2) == 0);
+    REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &ext3) == 0);
+
+    // Construct extent cache with three items.
+    ext2->next[0] = ext3;
+    ext->next[0] = ext2;
+    bms->ext_cache = ext;
+    bms->ext_cache_cnt = 3;
+    SECTION("Discard every item in extent list")
+    {
+        REQUIRE(__ut_block_ext_discard(session->getWtSessionImpl(), 0) == 0);
+        validate_and_free_ext_list(bms, 0);
+    }
+
+    SECTION("Discard until only one item is in extent list")
+    {
+        REQUIRE(__ut_block_ext_discard(session->getWtSessionImpl(), 1) == 0);
+
+        validate_and_free_ext_list(bms, 1);
+    }
+
+    SECTION("Discard nothing in the extent list")
```

Comment:
It took me a moment to match the section name ("Discard nothing in the extent list") with the parameter 3 passed to both `__ut_block_ext_discard()` and `validate_and_free_ext_list()`. Once I realised that the exit list had 3 items to start with, then that became clear.
I suggest either updating the section name, or adding a comment, to clarify that the 'discard nothing' is because the the number of items is already correct.

Reply:
Updated the section description [37eb6eb](https://github.com/wiredtiger/wiredtiger/pull/10963/commits/37eb6eb87910c7eae95b57d4d6bccdae2ca69fba)

Reply:
Thanks!